### PR TITLE
fix: standardise type of date/time object in the codebase

### PR DESCRIPTION
**Pull request by ryanMMM**
Refactored the codebase to exclusively use local date and local date time objects instead of alternating between local date time and zoned date time. Additionally, I refactored the expiry dates of prescriptions to be local date objects rather than local date time objects.

### CODE REVIEW

File: src/database/JsonDatabase.java [hunk 1]
```diff
@@ -53,10 +52,6 @@ public JsonDatabase(Class<T> type, KeyDelegator keyDelegator, File folder) {
 
         GsonBuilder builder = Converters.registerAll(new GsonBuilder());
 
-        // Unfortunately, Google's Gson library implicitly serializes immutable private objects which causes the
-        // compiler to issue a runtime warning. This is a temporary fix for now, until I find a better library.
-        builder.registerTypeAdapter(getZoneRegionClass(), new ZoneIdConverter());
-
```
Comment (ahmedkhalf) [inline src/database/JsonDatabase.java:59]: Wow, this was a huge pain point didn't realize this PR could fix it! Tnx a lot.

**Review by ahmedkhalf — APPROVED** [review]
LGTM!

**Comment by ahmedkhalf** [comment]
perhaps next time however look into making multiple commits. Your commit is too large.

**Comment by ryanMMM** [comment]
> perhaps next time however look into making multiple commits. Your commit is too large.

Zoned date time was used in many files so it required refactoring in a lot of files to ensure no bugs were pushed.

**Review by BaraKharseh — APPROVED** [review]

**Comment by ahmedkhalf** [comment]
> > perhaps next time however look into making multiple commits. Your commit is too large.
> 
> Zoned date time was used in many files so it required refactoring in a lot of files to ensure no bugs were pushed.

Oh I see, okay I guess this is understandable, especially since we don't wanna push uncompilable code.

**Pull request merged**